Home page

joeowenca opened the pull request:
Restructure home page with components and add more content

joeowenca committed:
Create introduction component

joeowenca committed:
Create showcase component

brettgoss commented on src/app/page.tsx:
```diff
@@ -1,23 +1,11 @@
+    <main>
+      <Introduction />
+      <Showcase />
     </main>
```
You're probably going to end up with a merge conflict here since your `header` branch is touching the same lines of code. Not a big deal, just something you'll have to resolve once you merge your header branch

brettgoss commented on src/app/components/showcase.tsx:
```diff
@@ -0,0 +1,15 @@
+      <h1 className="pb-3 text-center font-serif text-2xl">{"Photography"}</h1>
+      <h2 className="pb-3 font-serif text-xl">{"Sports"}</h2>
+      <h2 className="pb-3 font-serif text-xl">{"Automotive"}</h2>
+      <h2 className="pb-3 font-serif text-xl">{"Landscapes"}</h2>
```
You could probably get away with removing the `{"` since the text is just dynamic. In react, you use `{` for variables, and in this case you're basically saying "Inside this `<h2>` tag is a variable, and that variable is a string", and instead you should be able to just do:
```suggestion
      <h2 className="pb-3 font-serif text-xl">Landscapes</h2>
```

Not a huge deal either way, just cleans the code up a bit.

joeowenca replied:
Ended up cleaning this up while creating and adding the <ShowcaseItem> components

brettgoss commented on src/app/components/showcase.tsx:
```diff
@@ -0,0 +1,15 @@
+  return (
+    <div className="p-5 mt-[100vh] bg-black">
+      <h1 className="pb-3 text-center font-serif text-2xl">{"Photography"}</h1>
+      <h2 className="pb-3 font-serif text-xl">{"Sports"}</h2>
```
You have a decent chunk of repeated code here. Could be an opportunity to create a new `ShowcaseItem` component (in this same file). Then you could do something like 
```suggestion
      <ShowcaseItem>Sports</ShowcaseItem>
```
or if you don't want to use `children`, you could do
```suggestion
      <ShowcaseItem title="Sports" />
```

Another option is including the h1 in this new component so you could do 
```suggestion
      <ShowcaseItem title="Photography" menuItems={['Sports', 'Automotive', 'Landscapes']} />
```

joeowenca replied:
Thanks! Yeah, I found a lot of repeat code with the utility classes that I wasn't used to, but I've since refactored these into <ShowcaseItem> components where the title is accepted as a prop, and the description is added as children. This has been updated in a recent commit

brettgoss reviewed (approved):
Looks good! A few suggestions for how you could tidy up the code, but nothing major!

joeowenca committed:
Restructure showcase items as components

joeowenca committed:
Improve showcase text responsiveness

brettgoss commented on src/app/components/showcase.tsx:
```diff
@@ -0,0 +1,45 @@
+  )
+}
+
+interface ShowcaseItemProps {
```
When trying react props, it's generally preferably to use `type` instead of `interface`. Both are pretty similar, however `type` offers a bit of an easier to use syntax when the types get complex.

joeowenca replied:
Oh great, thanks! This was my first time working with prop types and TypeScript, so `interface` is just what I found online. I'll work on updating this!

joeowenca replied:
Done in a recent commit!

joeowenca committed:
Merge branch 'main' into home-page

joeowenca committed:
Position header at top-left of viewport

Upon merging branch 'main' into 'home-page', the header wasn't
displaying correctly. Changing the styling of the header to
top-0 and left-0 has resolved the issue.

joeowenca committed:
Change ShowcaseItemProps to use type instead of interface
